### PR TITLE
Server

### PR DESCRIPTION
**Pull request by engshorouq**
Relates #60 
## New feature
setup basic server by adding : 
 -  express
 - put middleware for cookie by cookie-parser
 - handle icon of website by serve-favicon
 - setup handlebars by express-handlebars
 - handle static files
##  Edits : 
- there is some review and fixed it

## commented line
- there are two commented lines in this pull request , when others in my team start working on the controller files then the should comment these line out

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+const controllers = require('./controllers/index');
+
+const app = express();
+app.set('port', process.env.PORT || 3300);
+app.disable('x-powered-by');
+app.use(serveFavicon(path.join(__dirname, '..', 'public', 'wallet.ico')));
+app.use(express.static(path.join(__dirname, '..', 'public')));
+app.set('views', path.join(__dirname, 'views'));
+app.use(express.json());
+app.use(express.urlencoded({ extended: false }));
```
Comment (IsraaSulaiman) [inline src/app.js]: I don't think we will use express.urlencoded in our project, @engshorouq

File: src/app.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+require('dotenv').config();
+const express = require('express');
+const expresshbs = require('express-handlebars');
+const path = require('path');
+const serveFavicon = require('serve-favicon');
+const cookieParser = require('cookie-parser');
+const controllers = require('./controllers/index');
```
Comment (IsraaSulaiman) [inline src/app.js]: It is a best practice to add a breaking line here to separate third-party packages and core modules from your own modules. Remember the issue? :D

**Review by KhaderMurtaja — APPROVED** [review]

**Review by mohammedmh — APPROVED** [review]

**Review by KhaderMurtaja — APPROVED** [review]

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+const app = require('./app');
+
+app.listen(app.get('port'), () => {
+  console.log(`server running on port ${app.get('port')}`);
```
Comment (asmaa-ezz) [inline src/index.js]: In the **index.js** file to use `app.get('port')`  twice
so I prefer to put the `app.get('port')`  in variable 
```
const port = app.get('port')
``` 
and use the variable whenever you want

File: src/app.js (hunk 2 above)
Comment (asmaa-ezz) [inline src/app.js]: **controllers** is assigned a value but never used.

**Review by asmaa-ezz — COMMENTED** [review]
Finally you can modify pull request
According to our Template

**Review by IsraaSulaiman — APPROVED** [review]

**Pull request merged**